app/jobs/my_modules/due_date_reminder_job.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class DueDateReminderJob < ApplicationJob
+  def perform
+    my_modules = MyModule.approaching_due_dates
+
+    my_modules.each do |task|
+      TaskDueDateNotification.send_notifications({ my_module_id: task.id })
+      task.update(notification_sent: true)
```

Comment:
move this notification_sent logic to the notification class, there should be some after_send hook or something like that, check noticed gem docs